Makefile

```diff
@@ -160,10 +164,11 @@ test: system
 toolchain:
 	@cd toolchain; ./toolchain-build.sh
 
+ASM_OBJS = kernel/boot.o kernel/gdt.o kernel/idt.o kernel/irq.o kernel/isr.o kernel/task.o kernel/tss.o kernel/user.o
```

Comment:
ASM_OBJS could actually be generated kernel/boot.o followed by a glob of .S:.o + filter-out of kernel/boot.o, would that work for you?